scripts/gen_istio_image_list.sh

```diff
@@ -0,0 +1,109 @@
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script generates the list of images used by Istio on the control
+# plane, data plane as well as addons.We generate this list by looking
+# at the values.yaml file where the Helm charts are stored, as well as
+# reading the values of DOCKER_TARGETS rule in the istio-docker.mk file.
+ 
+ISTIO_IMAGES_FILE=istio.IMAGES
+ISTIO_HELM_CHART_DIR="../install/kubernetes/helm/istio/charts/"
+
+function get_istio_images() {
+    buff=''
+    continue_processing=0
+    while IFS= read -r line 
+    do 
+       if [[ $line == DOCKER_TARGETS:=*\\ ]]
+       then
+           buff+=${line#DOCKER_TARGETS:=}
+           continue_processing=1
+       else
+           if [ $continue_processing -eq 1 ]
+           then
+              buff=${buff%\\}$line
+              break
+           fi
+       fi
+    done < ../tools/istio-docker.mk
+
+    for repo in $buff; do
+       istio_component=${repo##docker.};
+       istio_repo=istio/$istio_component;
+       istio_tag=$(get_latest_tag "$istio_repo")
+       echo docker.io/"$istio_repo":"$istio_tag" >> $ISTIO_IMAGES_FILE
+    done
+}
+
+function get_latest_tag() {
+    latest_tag=""
+    TAGS="`curl --silent 'https://registry.hub.docker.com/v2/repositories/'"$1"'/tags/' | grep -o '"name": "[0-9].[0-9].[0-9]"' | tr -d '"' `"
+    for tag in $TAGS;
+    do
+      if [[ $tag != name: ]]
+      then
+         if [[ $tag > $latest_tag ]]
+         then
+            latest_tag=$tag
+         fi
+      fi
+    done
+    echo "$latest_tag"
+}
+
+function get_istio_addons() {
+    for dir in "$ISTIO_HELM_CHART_DIR"*
+    do
+        if [[ -d $dir ]]
+        then
+           if [[ -f "$dir"/values.yaml  ]]
+           then
+              HUB=$(cat "$dir"/values.yaml | grep -E 'hub: |repository: ' | awk '{print $2 $4}')
+              IMAGE=$(cat "$dir"/values.yaml | grep "image: " | awk '{print $2 $4}')
+              TAG=$(grep "tag: " "$dir"/values.yaml | awk '{print $2 $4}')
+
+              hubs=()
+	      images=()
+	      tags=()
+
+              for hub in $HUB;
+                 do hubs+=( "$hub" );
+              done
+
+	      for img in $IMAGE;
+                 do images+=( "$img" );
+              done
+
+              for tag in $TAG;
+                 do tags+=( "$tag" );
+              done
+
+              for i in "${!hubs[@]}"; do
+                  if [ -z "${images[$i]}" ]
+                  then
+                    echo "${hubs[$i]}":"${tags[$i]}" >> $ISTIO_IMAGES_FILE
+                  else
+                    echo "${hubs[$i]}"/"${images[$i]}":"${tags[$i]}" >> $ISTIO_IMAGES_FILE
```

Comment:
This should just write to stdout probably, then it is more flexible

Reply:
Done.